copy.js

```diff
@@ -8,8 +8,39 @@
 
 var global_link_text = '';
 
+function maybe_clean (orig_url, regexp, f) {
+    var res = '';
+    for (var pat in regexp) {
+        if (new RegExp(pat).test(orig_url)) {
+            res = link_cleaner(orig_url, f)['redirectUrl'];
+            break;
+        }
+    }
+    return (res != '' ? res : orig_url);
+}
+
+function cleaner_entrypoint (orig_link) {
```

Comment:
And this is part that hurts.

In order to respond to user action (requesting to clean a URL), I'm basically replicating what the browser does in `background.js` - matching the URL and decide to apply or not a certain cleaning function.

I guess, though, that running so much regexping like this hitting hard on performance, but this is just a proof of my proposal for the feature (implementing the cleaning action on the menu item). If the idea is sound and makes sense, we can improve here and try to reduce the memory footprint.

Reply:
Yeah this does not look great. I guess on the bright side, it's only doing it when you try to copy a link?

Reply:
correct